common/src/app/common/types/file.cljc

```diff
@@ -1149,3 +1151,65 @@
               (validate-shape-not-component shape file page libraries report-error))))))
 
     @errors))
+
+;; Export
+
+(defn- get-component-ref-file
+  [objects shape]
+
+  (cond
+    (contains? shape :component-file)
+    (get shape :component-file)
+
+    (contains? shape :shape-ref)
+    (recur objects (get objects (:parent-id shape)))
+
+    :else
+    nil))
+
+(defn detach-external-references
+  [file file-id]
+  (let [detach-text
+        (fn [content]
+          (->> content
+               (ct/transform-nodes
+                #(cond-> %
+                   (not= file-id (:fill-color-ref-file %))
+                   (dissoc :fill-color-ref-id :fill-color-ref-file)
+
+                   (not= file-id (:typography-ref-file %))
+                   (dissoc :typography-ref-id :typography-ref-file)))))
+
+        detach-shape
+        (fn [objects shape]
+          (l/info :msg "detach-shape")
+          (l/info :msg file-id)
+          (l/info :msg (get-component-ref-file objects shape))
```

Comment:
Please, use a single log line, in the same style as it is used in the backend binfile ns
The `info` logging level also looks incorrect, it should be probably `debug` or `trace` (check the binfile ns and use it in the same way)

Reply:
Fixed